chess/src/Components/PlayOnline.js

```diff
@@ -83,37 +98,92 @@ class PlayOnline extends React.Component {
   }
 
   async onDrop(sourceSquare, targetSquare) {
-    console.log(this.state.players);
-    const newGame = new Chess(this.state.game);
-    const gameCopy = { ...newGame };
-    const move = gameCopy.move({
-      from: sourceSquare,
-      to: targetSquare,
-      promotion: "q", // always promote to a queen for example simplicity
-    });
-
-    if (move === null) return false;
-    const newGameFen = gameCopy.fen();
-
-    this.socket.emit("new move", newGameFen);
-    this.setState({ game: newGameFen });
-
-    if (!this.state.endOfGame && gameCopy.in_checkmate()) {
-      let opponentName = this.getOpponentName();
-      this.setState({ endOfGame: true, message: "You are the winner!" });
-      await this.sendResults(this.props.userData.user, 1, 0, 0);
-      await this.sendResults(opponentName, 0, 1, 0);
-      this.socket.emit("new message", `${this.props.userData.user} Wins`);
+    if (
+      !this.state.endOfGame &&
+      this.props.userData.user ===
+        this.state.players[this.state.moveCounter % 2] &&
+      this.state.players.length === 2
+    ) {
+      const newGame = new Chess(this.state.game);
+      const gameCopy = { ...newGame };
+      const move = gameCopy.move({
+        from: sourceSquare,
+        to: targetSquare,
+        promotion: "q", // always promote to a queen for example simplicity
+      });
+
+      if (move === null) return false;
+      const newGameFen = gameCopy.fen();
+
+      this.socket.emit("new move", newGameFen);
+      // const moveIncrement = this.state.moveCounter + 1;
+      this.setState({ game: newGameFen });
+
+      if (!this.state.endOfGame && gameCopy.in_checkmate()) {
```

Comment:
Maybe we could think about sending the results in the form of: winFunction(this.state.players[0]), loseFunction(this.state.players[1])